up_client_zenoh/upclientzenoh.py

```diff
@@ -0,0 +1,463 @@
+                msg = "Unable to get payload encoding"
+                print(msg)
+                return UStatus(code=UCode.INTERNAL, message=msg)
+
+        # Create Zenoh subscriber
+        try:
+
+            subscriber = self.session.declare_subscriber(zenoh_key, callback)
+            if subscriber:
+                with self.subscriber_lock:
+                    self.subscriber_map[(topic.SerializeToString(), listener)] = subscriber
+
+            else:
+                msg = "Unable to register callback with Zenoh"
+                print(msg)
+                return UStatus(code=UCode.INTERNAL, message=msg)
+        except Exception as e:
+            msg = "Unable to register callback with Zenoh"
+            print(msg)
+            return UStatus(code=UCode.INTERNAL, message=msg)
+        msg = "Successfully register callback with Zenoh"
+        print(msg)
+        return UStatus(code=UCode.OK, message=msg)
+
+    def register_request_listener(self, topic: UUri, listener: UListener) -> UStatus:
+        zenoh_key = ZenohUtils.to_zenoh_key_string(topic)
+
+        def callback(query: Query) -> None:
+            nonlocal self, listener, topic
+            attachment = query.attachment
+            if not attachment:
+                msg = "Unable to get attachment"
+                print(msg)
+                return UStatus(code=UCode.INTERNAL, message=msg)
+
+            u_attribute = ZenohUtils.attachment_to_uattributes(attachment)
+            if isinstance(u_attribute, UStatus):
+                msg = f"Unable to transform user attachment to UAttributes: {u_attribute}"
+                print(msg)
+                return UStatus(code=UCode.INTERNAL, message=msg)
+
+            value = query.value
+            if value:
+                encoding = ZenohUtils.to_upayload_format(value.encoding)
+                if not encoding:
+                    msg = "Unable to get payload encoding"
+                    print(msg)
+                    return UStatus(code=UCode.INTERNAL, message=msg)
+
+                u_payload = UPayload(format=encoding, value=value.payload)
+            else:
+                u_payload = UPayload(format=UPayloadFormat.UPAYLOAD_FORMAT_UNSPECIFIED)
+
+            msg = UMessage(attributes=u_attribute, payload=u_payload)
+
+            self.query_map[u_attribute.id.SerializeToString()] = query
+            listener.on_receive(msg)
+
+        def declare_queryable(callback: Any) -> Any:
+            nonlocal self, zenoh_key, topic, listener
+            try:
+                queryable = self.session.declare_queryable(zenoh_key, callback)
+                with self.queryable_lock:
+                    self.queryable_map[(topic.SerializeToString(), listener)] = queryable
+            except Exception as e:
+                msg = "Unable to register callback with Zenoh"
+                print(msg)
+                return UStatus(code=UCode.INTERNAL, message=msg)
+
+        declare_queryable(callback)
```

Comment:
Why do we need to create a local function and call it here?
Can't we run what we have in the function directly?

Reply:
Removed since it didn't add any value, as this local function is only used once